service/topology_coordinator.cc

```diff
@@ -2812,6 +2861,26 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    future<> maybe_start_view_building_coordinator() {
+        if (!_feature_service.view_building_coordinator) {
+            co_return;
+        }
```

Comment:
We should also make sure that the view build coordinator is started when the feature becomes enabled. Right now, `maybe_start_view_building_coordinator` will start it if the current node _becomes_ a group0 leader and the feature is already enabled, but it might happen that the feature becomes enabled in the middle of a leader's term.